### PR TITLE
Toolchains to download prebuilt cmake and ninja

### PR DESCRIPTION
**Pull request by jsharpe**

### CODE REVIEW

**Comment by jsharpe** [comment]
Still need to fix up ninja on aarch64 as that's not currently correct.

File: examples/deps/repositories.bzl [hunk 1]
```diff
@@ -1,7 +1,17 @@
 # buildifier: disable=module-docstring
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-all_content = """filegroup(name = "all", srcs = glob(["**"]), visibility = ["//visibility:public"])"""
+all_content = """filegroup(name = "all", srcs = glob(["**"], exclude=["WORKSPACE", "WORKSPACE.bazel", "BUILD", "BUILD.bazel"]), visibility = ["//visibility:public"])"""
```
Comment (UebelAndre) [inline examples/deps/repositories.bzl:4]: Nit: Can you update these to be a bit more expanded
```python
_ALL_CONTENT = """\
filegroup(
    name = "all_srcs", 
    srcs = glob(
        [
            "**"
        ], 
        exclude=[
            "WORKSPACE", 
            "WORKSPACE.bazel", 
            "BUILD", 
            "BUILD.bazel",
        ],
    ),
)
"""

_ALL_CONTENT_CMAKE= """\
load("@rules_foreign_cc//for_workspace:cmake_build.bzl", "cmake_tool")

package(default_visibility = ["//visibility:public"])

filegroup(
    name = "all_srcs", 
    srcs = glob(
        [
            "**"
        ], 
        exclude=[
            "WORKSPACE", 
            "WORKSPACE.bazel", 
            "BUILD", 
            "BUILD.bazel",
        ],
    ),
)

cmake_tool(
    name = "cmake", 
    cmake_srcs=":all_srcs",
)
"""


_ALL_CONTENT_NINJA = """\
load("@rules_foreign_cc//for_workspace:ninja_build.bzl", "ninja_tool")

package(default_visibility = ["//visibility:public"])

filegroup(
    name = "all_srcs", 
    srcs = glob(
        [
            "**"
        ], 
        exclude=[
            "WORKSPACE", 
            "WORKSPACE.bazel", 
            "BUILD", 
            "BUILD.bazel",
        ],
    ),
)

ninja_tool(
    name = "ninja", 
    ninja_srcs=":all_srcs",
)
"""
```

File: examples/deps/repositories.bzl [hunk 2]
```diff
@@ -14,6 +24,70 @@ def repositories():
         type = "zip",
     )
 
+    http_archive(
```
Comment (UebelAndre) [inline examples/deps/repositories.bzl:27]: Nit: I feel like this should be something like `cmake_src`

File: examples/deps/repositories.bzl [hunk 3]
```diff
@@ -14,6 +24,70 @@ def repositories():
...
+        build_file_content = all_content,
+    )
+
+    http_archive(
+        name = "cmake_macos_universal",
+        sha256 = "eb1f52996632c1e71a1051c9e2c30cc8df869fb5a213b1a0d3b202744c6c5758",
+        urls = [
+            "https://cmake.org/files/v3.19/cmake-3.19.4-macos-universal.tar.gz",
+        ],
+        strip_prefix = "cmake-3.19.4-macos-universal",
```
Comment (UebelAndre) [inline examples/deps/repositories.bzl:61]: Nit: Would it not be better to also strip `CMake.app/Contents`?

**Review by UebelAndre — APPROVED** [review]
A few nits but otherwise looks good for now! I think ideally there should be a way to specify what versions they care about but this is far better than having broken CI 😄 

Thank you so much!!

File: examples/deps/repositories.bzl [hunk 4]
```diff
@@ -1,7 +1,17 @@
 # buildifier: disable=module-docstring
```
Comment (UebelAndre) [inline examples/deps/repositories.bzl:1]: Wait, shouldn't the changes in this file go in `workspace_definitions.bzl` or something?

**Review by UebelAndre — CHANGES_REQUESTED** [review]

File: examples/deps/repositories.bzl (hunk 3 above)
Comment (jsharpe) [reply examples/deps/repositories.bzl]: By having the `CMake.app` in the folder structure this causes macOS to treat this as a application bundle. I don't know if removing this from the folder name breaks anything?

**Comment by jsharpe** [comment]
@UebelAndre PTAL

File: examples/deps/repositories.bzl (hunk 3 above)
Comment (UebelAndre) [reply examples/deps/repositories.bzl]: Bazel is never going to execute it as an application bundle. I know from experience that you can remove this.

**Comment by jsharpe** [comment]
Closing in favour of #508

**Pull request closed without merging**